src/libraries/Microsoft.PowerFx.Connectors/Public/ConnectorTableValue.cs

```diff
@@ -34,5 +36,32 @@ internal ConnectorTableValue(IRContext irContext)
         public virtual void Refresh()
         {
         }
+
+        public async Task<IReadOnlyCollection<DValue<RecordValue>>> GetRowsAsync(IServiceProvider services, DelegationParameters parameters, CancellationToken cancel)
+        {
+            var op = parameters.ToOdataParameters();
+            var rows = await _tabularService.GetItemsAsync(services, op, cancel).ConfigureAwait(false);
+
+            return rows;
+        }
+    }   
+
+    internal static class ODataParametersExtensions
+    {
+        public static ODataParameters ToOdataParameters(this DelegationParameters parameters)
```

Comment:
Could we eventually get rid of ODataParameters and just use DelegationParameters natively?